### PR TITLE
[Task#10026]User detail screen

### PR DESCRIPTION
**Pull request by chuongmd-1771**
### Issue Link :link:

https://edu-redmine.sun-asterisk.vn/issues/10026
### Goals :soccer:

- Design UI User Detail Screen

### CODE REVIEW

**Comment by chuongmd-1771** [comment]
ready

File: Petty/Scenes/Home/HomeViewController.swift [hunk 1]
```diff
@@ -10,10 +10,15 @@ import UIKit
...
 final class HomeViewController: UIViewController {
-    
+
     override func viewDidLoad() {
         super.viewDidLoad()
     }
+    
+    @IBAction func handleGotoUserDetail(_ sender: Any) {
+        let vc = UserDetailViewController.instantiate()
+        self.present(vc, animated: true)
```
Comment (tienpx3010) [inline Petty/Scenes/Home/HomeViewController.swift]:
```suggestion
        present(vc, animated: true)
```

File: Petty/Scenes/User Detail/UserDetailViewController.swift [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+//
+//  Created by chuongmd on 4/10/19.
+//  Copyright © 2019 Sun-Asterisk. All rights reserved.
+//
+
+import UIKit
+import Reusable
+
+final class UserDetailViewController: UIViewController {
+    @IBOutlet weak var stackView: UIView!
```
Comment (tienpx3010) [inline Petty/Scenes/User Detail/UserDetailViewController.swift]: private

**Review by ducnm-1825 — APPROVED** [review]

**Comment by chuongmd-1771** [comment]
ready

**Comment by chuongmd-1771** [comment]
ready

**Comment by chuongmd-1771** [comment]
ready

File: Petty/Extensions/UIView+.swift [hunk 3]
```diff
@@ -21,7 +25,10 @@ extension UIView {
         layer.rasterizationScale = scale ? UIScreen.main.scale : 1
     }
 
-    func roundCorner(color: UIColor, cornerRadius: CGFloat, shadowRadius: CGFloat, clipsToBounds: Bool) {
+    func roundCorner(color: UIColor,
+                     cornerRadius: CGFloat,
+                     shadowRadius: CGFloat,
+                     clipsToBounds: Bool) {
         self.clipsToBounds = true
```
Comment (tienpx3010) [inline Petty/Extensions/UIView+.swift]:
```suggestion
        clipsToBounds = true
```

File: Petty/Extensions/UIView+.swift [hunk 4]
```diff
@@ -53,3 +62,50 @@ extension UIView {
...
+        
+        addConstraints(NSLayoutConstraint.constraints(withVisualFormat: format,
+                                                      options: NSLayoutConstraint.FormatOptions(),
+                                                      metrics: nil,
+                                                      views: viewsDictionary))
+    }
+}
+
+extension UIView {
+    @IBInspectable var cornerRadiusV: CGFloat {
```
Comment (tienpx3010) [inline Petty/Extensions/UIView+.swift]:
```suggestion
    @IBInspectable var cornerRadius: CGFloat {
```
Tương tự phía dưới

File: Petty/Scenes/My Photos/MyPhotosCollectionViewController.swift [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+        return itemInfo
+    }
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+//        guard collectionView != nil else { return }
+//        self.collectionView.register(UICollectionViewCell.self, forCellWithReuseIdentifier: reuseIdentifier)
+    }
+//
+//    override func numberOfSections(in collectionView: UICollectionView) -> Int {
```
Comment (tienpx3010) [inline Petty/Scenes/My Photos/MyPhotosCollectionViewController.swift]: Xoá

File: Petty/Scenes/My Photos/MyPhotosCollectionViewController.swift [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+//
+
+import UIKit
+import XLPagerTabStrip
+
+private let reuseIdentifier = "Cell"
+
+final class MyPhotos: UITableViewController, IndicatorInfoProvider {
+    
+    var itemInfo = IndicatorInfo(title: Constant.itemInfoTitle)
```
Comment (tienpx3010) [inline Petty/Scenes/My Photos/MyPhotosCollectionViewController.swift]: private

File: Petty/Scenes/My Photos/MyPhotosCollectionViewController.swift [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+//  Petty
+//
+//  Created by chuongmd on 4/11/19.
+//  Copyright © 2019 Sun-Asterisk. All rights reserved.
+//
+
+import UIKit
+import XLPagerTabStrip
+
+private let reuseIdentifier = "Cell"
```
Comment (tienpx3010) [inline Petty/Scenes/My Photos/MyPhotosCollectionViewController.swift]: Bỏ vào enum Constants trong viewcontroller

File: Petty/Scenes/User Detail/UserDetailViewController.swift [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+//  Copyright © 2019 Sun-Asterisk. All rights reserved.
+//
+
+import UIKit
+import Reusable
+import XLPagerTabStrip
+
+final class UserDetailViewController: BaseButtonBarPagerTabStripViewController<MenuBarButtonCell> {
+    @IBOutlet weak var topView: UIView!
+    @IBOutlet weak var avatarImage: UIImageView!
```
Comment (tienpx3010) [inline Petty/Scenes/User Detail/UserDetailViewController.swift]: private

File: Petty/Extensions/UIView+.swift (hunk 3 above)
Comment (chuongmd-1771) [reply Petty/Extensions/UIView+.swift]: Cannot assign to value: 'clipsToBounds' is a 'let' constant
Add explicit 'self.' to refer to mutable property of 'UIView'

Xcode báo phải để self

**Comment by chuongmd-1771** [comment]
ready

**Comment by chuongmd-1771** [comment]
ready

**Review by tienpx3010 — APPROVED** [review]

**Pull request merged**